### PR TITLE
[WIP] Migration to `noetic`

### PR DESCRIPTION
**Pull request by mhwasil**
* Change pcl library to 1.10
* Change opencv to 4.x.x.
* Use boost to generate random for point sampling
* Use cmake 3.x.x

### CODE REVIEW

**Comment by sthoduka** [comment]
can you run the linter? all the code uses 2 spaces now instead of 4.  I think we can merge it immediately and then continue pushing changes as required. That would make it easier for others to push to noetic as well

File: mir_perception/mir_perception_utils/CMakeLists.txt [hunk 1]
```diff
@@ -18,9 +14,12 @@ find_package(catkin REQUIRED
     visualization_msgs
 )
 
-find_package(PCL 1.7 REQUIRED)
+find_package(PCL 1.10 REQUIRED)
 find_package(VTK REQUIRED)
 find_package(OpenCV REQUIRED)
+#find_package(Boost REQUIRED python)
```
Comment (sthoduka) [inline mir_perception/mir_perception_utils/CMakeLists.txt:20]: we can remove these commented lines

File: mir_perception/mir_perception_utils/common/src/pointcloud_utils.cpp [hunk 2]
```diff
@@ -51,41 +51,47 @@ unsigned int pointcloud::centerPointCloud(const PointCloud &cloud_in, PointCloud
...
+    if (point_count > num_points) 
+    {
+        unsigned int point_diff = point_count - num_points;
 
-  if (point_count > num_points) {
-    unsigned int point_diff = point_count - num_points;
+        boost::random::mt19937 rd; 
+        boost::random::uniform_int_distribution<> distr(0,point_count);
 
-    std::random_device rd;   // get ran
```
Comment (sthoduka) [inline mir_perception/mir_perception_utils/common/src/pointcloud_utils.cpp:62]: this doesn't work with c++14?

Comment (mhwasil) [reply mir_perception/mir_perception_utils/common/src/pointcloud_utils.cpp:62]: No, it was not working. That's why I changed it to boost

**Comment by sthoduka** [comment]
I've cherry-picked some of the commits from here and pushed to noetic branch

**Pull request closed without merging**